src/legacy/core_plugins/visualizations/public/np_ready/public/types.ts

```diff
@@ -18,21 +18,33 @@
  */
 
 import { SavedObject } from '../../../../../../plugins/saved_objects/public';
-import { Vis, VisState, VisParams, VisualizationController } from './vis';
-import { ISearchSource } from '../../../../../../plugins/data/public/';
-import { SavedSearch } from '../../../../../../plugins/discover/public';
+import { ISearchSource, AggConfigOptions } from '../../../../../../plugins/data/public';
+import { SerializedVis, Vis, VisParams } from './vis';
 
-export { Vis, VisState, VisParams, VisualizationController };
+export { Vis, SerializedVis, VisParams };
 
-export interface VisSavedObject extends SavedObject {
-  vis: Vis;
-  description?: string;
-  searchSource: ISearchSource;
+export interface VisualizationController {
+  render(visData: any, visParams: any): Promise<void>;
+  destroy(): void;
+  isLoaded?(): Promise<void> | void;
+}
+
+export interface SavedVisState {
+  type: string;
```

Comment:
The `title` should be presented here as it's also presented in the url and can be changed through the url.
There is also an enhancement request for handling the title from the url:  https://github.com/elastic/kibana/issues/57100

Reply:
this is the type of the vis as we save it in .kibana index (not as we have it in the url)
title is present on ISavedVis, which also has visState (which has aggs, params and type, but no title)

Reply:
i added title to PureVisState editor is using